bin/travis-encrypt-file

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+
+set -euo pipefail # STRICT MODE
+IFS=$'\n\t'       # http://redsymbol.net/articles/unofficial-bash-strict-mode/
+
+random-hex() {
+  head -c 10 /dev/urandom | xxd -p
+}
+
+NC='\033[0m'
+
+plain() {
+  echo -en $* 1>&2
+}
+
+blue() {
+  echo -en "\033[0;34m$*${NC}" 1>&2
+}
+
+yellow() {
+  echo -en "\033[0;33m$*${NC}" 1>&2
+}
+
+red() {
+  echo -en "\033[0;31m$*${NC}" 1>&2
+}
+
+gray() {
+  echo -en "\033[0;37m$*${NC}" 1>&2
+}
+
+nl() {
+  echo 1>&2
+}
+
+if ! git config travis.slug &>/dev/null; then
+  plain "Please define the "
+  yellow "travis.slug"
+  plain " value using "
+  yellow "git config"
```

Comment:
`git config --local`, preferably... Maybe run `git config --local --add travis.slug $(git remote get-url upstream | sed -Ee 's/.*:(.*).git/\1/')` to auto-set it?

Reply:
I thought about auto-setting, but I didn't want to go too crazy. This tool needs the `travis` CLI anyway, so it's easy enough to delegate it to that. Also `git-config` defaults to `--local`, though we could add it just to be explicit.